src/app/ngrx/reducers/episode-downloads.reducer.ts

```diff
@@ -29,32 +29,53 @@ export function reducer(state: State = initialState, action: AllActions) {
     case ActionTypes.CASTLE_EPISODE_DOWNLOADS_LOAD: {
       const { guid, podcastId, page } = action.payload;
       return adapter.upsertOne(
-        {id: guid, changes: { guid, podcastId, page, error: null, loading: true, loaded: false}},
+        {
+          id: guid,
+          ...selectEpisodeDownloadsEntities(state)[guid],
```

Comment:
Since upsert doesn't take `changes` anymore, I tend to just prefill it with the current record when I'm not intending to set all the values with an action.